### PR TITLE
Fix molecule and atom renaming

### LINKED ISSUE #524: Parameterize: preserve original atom names
Currently `parameterize` renames atoms to comply with the `match` requirements, though this should be hidden from a user and original names used though out the tool.

### LINKED ISSUE #557: CGENFF atom typing issues: KeyError: 'HO1'
MOL2 file https://isu.box.com/s/1mi3sjt4aemaiadypigna8to01jgqnmc
Command used : parameterize 5TUD_Prepped_ERM.mol2 -c 1 -n 4 --forcefield CGENFF -o 5TUD_Prepped_ERM 

 === Fitting for CGenFF_2b6 ===

Traceback (most recent call last):
  File "/garlic/apps/anaconda3/envs/htmd/bin/parameterize", line 11, in <module>
    sys.exit(main_parameterize())
  File "/garlic/apps/anaconda3/envs/htmd/lib/python3.6/site-packages/htmd/parameterization/cli.py", line 186, in main_parameterize
    outdir=args.outdir)
  File "/garlic/apps/anaconda3/envs/htmd/lib/python3.6/site-packages/htmd/parameterization/ffmolecule.py", line 93, in __init__
    self.atomtype[:] = [self._rtf.type_by_name[name] for name in self.name]
  File "/garlic/apps/anaconda3/envs/htmd/lib/python3.6/site-packages/htmd/parameterization/ffmolecule.py", line 93, in <listcomp>
    self.atomtype[:] = [self._rtf.type_by_name[name] for name in self.name]
KeyError: 'HO1'

### LINKED ISSUE #643: parameterize resname error in rtf
The resname of the ligand is modified by parameterize when it compute CGENFF parameters and topology,

orig-mol2: ligname --> 0YA
mol.rtf: ligname --> MOL

### PR DESCRIPTION
**Pull request by raimis**
- [x] Enforce unique atom names
- [x] Clean up element guessing
- [x] Move atom renaming to `fftype` internals
  - [x] Update documentation
- [x] Update tests
- [x] Fix #524 
- [x] Fix #557
- [x] Fix #643 

This is a remake of #781 

Depend on https://github.com/Acellera/htmd-data/pull/10

### CODE REVIEW

**Comment by raimis** [comment]
@j3mdamas can you review it again!

**Comment by raimis** [comment]
@j3mdamas, the merge conflicts has been resolved!

File: htmd/parameterization/cli.py [hunk 1]
```diff
@@ -70,6 +70,35 @@ def getArgumentParser():
...
+
+    mol = Molecule(args.filename, guessNE=['bonds'], guess=[])
+
+    # Check if the file contain just one conformation
+    if mol.numFrames != 1:
+        raise RuntimeError('{} has to contain only one molecule, but found {}'.format(args.filename, mol.numFrames))
+
+    # Check if each atom name is unique
+    if np.unique(mol.name).size != mol.numAtoms:
+        raise RuntimeError('The atom names in {} has to be unique!'.format(args.filename))
```
Comment (j3mdamas) [inline htmd/parameterization/cli.py]: It's fine for now, but can be annoying for a user that wants to parameterize a molecule that may have atoms with the same name.

File: htmd/parameterization/fftype.py [hunk 2]
```diff
@@ -3,31 +3,52 @@
...
-import logging
-import unittest
 
 logger = logging.getLogger(__name__)
 
 
 fftypemethods = ('CGenFF_2b6', 'GAFF', 'GAFF2')
 
 
-def listFftypemethods():
```
Comment (j3mdamas) [inline htmd/parameterization/fftype.py]: I continue to see no reason to delete this, but ok

File: htmd/parameterization/cli.py (hunk 1 above)
Comment (raimis) [reply htmd/parameterization/cli.py]: This is address with by https://github.com/Acellera/htmd/pull/793

File: htmd/parameterization/fftype.py (hunk 2 above)
Comment (raimis) [reply htmd/parameterization/fftype.py]: What is the reason to keep it?

**Comment by stefdoerr** [comment]
Doesn't matter. At this point we don't plan to support outright a
programming interface for it anyway

File: htmd/parameterization/util.py [hunk 3]
```diff
@@ -30,93 +31,55 @@ def getEquivalentsAndDihedrals(mol):
...
-                return 'Cl'
-            else:
-                return 'C'
-        else:
-            raise ValueError('Cannot create element from atom name: {}. It probably does not match the atom elements'
-                             'available for {}: {}'.format(name, fftypemethod, elements[fftypemethod]))
-    else:
-        raise RuntimeError('Not a valid fftypemethod: {}. Valid methods: {}'.format(fftypemethod,
-                                                                                    ','.join(fftypemethods)))
+                raise RuntimeError('No chemical bonds found in the molecule')
```
Comment (j3mdamas) [inline htmd/parameterization/util.py]: I thought warning and guessing bonds was nice, but OK

**Pull request merged**